src/utils.ts

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import moment = require('moment');
+
+
+/**
+ * Wrappable Error class.
+ *
+ * The Error class doesn't actually act on `this`.  Instead it always
+ * returns a new instance of Error.  Here we capture that instance so we
+ * can apply it's properties to `this`.
+ */
+export
+class WrappedError implements Error {
```

Comment:
should we inherit from Error here? It seems odd that we are applying its constructor when we don't have its prototype in the prototype chain. But Error is one of those weird edge cases...